packages/contracts/contracts/test/DepositFeed.t.sol

```diff
@@ -0,0 +1,126 @@
+
+        df.depositTransaction(NON_ZERO_ADDRESS, ZERO_VALUE, NON_ZERO_GASLIMIT, false, NON_ZERO_DATA);
+    }
+
+    // Test: depositTransaction should emit the correct log when an EOA deposits a contract creation with 0 value
+    function test_depositTransaction_createWithZeroValueForEOA() external {
+        // EOA emulation
+        vm.prank(address(this), address(this));
+ 
+        vm.expectEmit(true, true, false, true);
+        emit TransactionDeposited(address(this), ZERO_ADDRESS, ZERO_VALUE, ZERO_VALUE, NON_ZERO_GASLIMIT, true, NON_ZERO_DATA);
+
+        df.depositTransaction(ZERO_ADDRESS, ZERO_VALUE, NON_ZERO_GASLIMIT, true, NON_ZERO_DATA);
+    }
+
+    // Test: depositTransaction should emit the correct log when a contract deposits a contract creation with 0 value
+    function test_depositTransaction_createWithZeroValueForContract() external {
+        vm.expectEmit(true, true, false, true);
+        emit TransactionDeposited(addressAlias(), ZERO_ADDRESS, ZERO_VALUE, ZERO_VALUE, NON_ZERO_GASLIMIT, true, NON_ZERO_DATA);
+
+        df.depositTransaction(ZERO_ADDRESS, ZERO_VALUE, NON_ZERO_GASLIMIT, true, NON_ZERO_DATA);
+    }
+
+    // Test: depositTransaction should increase its eth balance when an EOA deposits a transaction with ETH
+    function test_depositTransaction_withEthValueFromEOA() external {
+        // EOA emulation
+        vm.prank(address(this), address(this));
+
+        vm.expectEmit(true, true, false, true);
+        emit TransactionDeposited(address(this), NON_ZERO_ADDRESS, NON_ZERO_VALUE, ZERO_VALUE, NON_ZERO_GASLIMIT, false, '0x');
+
+        uint256 balanceBefore = address(df).balance;
+        df.depositTransaction{value: NON_ZERO_VALUE}(NON_ZERO_ADDRESS, ZERO_VALUE, NON_ZERO_GASLIMIT, false, '0x');
+        uint256 balanceAfter = address(df).balance;
+        assertEq(balanceAfter - balanceBefore, NON_ZERO_VALUE);
+    }
+
+    // Test: depositTransaction should increase its eth balance when a contract deposits a transaction with ETH
+    function test_depositTransaction_withEthValueFromContract() external {
+        vm.expectEmit(true, true, false, true);
+        emit TransactionDeposited(addressAlias(), NON_ZERO_ADDRESS, NON_ZERO_VALUE, ZERO_VALUE, NON_ZERO_GASLIMIT, false, NON_ZERO_DATA);
+
+        df.depositTransaction{value: NON_ZERO_VALUE}(NON_ZERO_ADDRESS, ZERO_VALUE, NON_ZERO_GASLIMIT, false, NON_ZERO_DATA);
+    }
+
+    // Test: depositTransaction should increase its eth balance when an EOA deposits a contract creation with ETH
+    function test_depositTransaction_withEthValueAndEOAContractCreation() external {
+        // EOA emulation
+        vm.prank(address(this), address(this));
+
+        vm.expectEmit(true, true, false, true);
+        emit TransactionDeposited(address(this), ZERO_ADDRESS, NON_ZERO_VALUE, ZERO_VALUE, NON_ZERO_GASLIMIT, true, '0x');
+
+        uint256 balanceBefore = address(df).balance;
```

Comment:
Since you're using the `setup()` function, which is basically like `beforeEach()`, I think that this will always be zero, so the line could be removed and would clean up the assertion as well (same for subsequent test).